packages/shared/lib/services/file/local.service.ts

```diff
@@ -1,32 +1,43 @@
-import type { Response } from 'express';
 import fs from 'fs';
 import path, { dirname } from 'path';
 import { fileURLToPath } from 'url';
+
 import archiver from 'archiver';
-import errorManager, { ErrorSourceEnum } from '../../utils/error.manager.js';
-import { NangoError } from '../../utils/error.js';
-import { LogActionEnum } from '../../models/Telemetry.js';
+
 import { nangoConfigFile } from '@nangohq/nango-yaml';
 
+import { LogActionEnum } from '../../models/Telemetry.js';
+import { NangoError } from '../../utils/error.js';
+import errorManager, { ErrorSourceEnum } from '../../utils/error.manager.js';
+
+import type { DBSyncConfig, NangoProps } from '@nangohq/types';
+import type { Response } from 'express';
+
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = dirname(__filename);
 
-const SYNC_FILE_EXTENSION = 'js';
+const scriptTypeToPath: Record<NangoProps['scriptType'], string> = {
+    'on-event': 'on-events',
+    action: 'actions',
+    sync: 'syncs',
+    webhook: 'syncs'
+};
+
+const basePath = process.env['NANGO_INTEGRATIONS_FULL_PATH'] || path.resolve(__dirname, `../nango-integrations`);
 
 class LocalFileService {
-    public getIntegrationFile(syncName: string, providerConfigKey: string, setIntegrationPath?: string | null) {
+    public getIntegrationFile({
+        scriptType,
+        syncConfig,
+        providerConfigKey
+    }: {
+        scriptType: NangoProps['scriptType'];
+        syncConfig: DBSyncConfig;
+        providerConfigKey: string;
+    }) {
         try {
-            const filePath = setIntegrationPath ? `${setIntegrationPath}dist/${syncName}.${SYNC_FILE_EXTENSION}` : this.resolveIntegrationFile(syncName);
-            const fileNameWithProviderConfigKey = filePath.replace(`.${SYNC_FILE_EXTENSION}`, `-${providerConfigKey}.${SYNC_FILE_EXTENSION}`);
-
-            let realPath;
-            if (fs.existsSync(fileNameWithProviderConfigKey)) {
-                realPath = fs.realpathSync(fileNameWithProviderConfigKey);
-            } else {
-                realPath = fs.realpathSync(filePath);
-            }
```

Comment:
SImplified this logic because providerConfigKey was added more than a year ago d909250057e390e2c468348d9060553cf0b17514